### PR TITLE
[SupportRP] Az.Support PowerShell module for Support RP API stable version "2020-04-01"

### PR DESCRIPTION
**Pull request by shahbj79**
## Description

- This is the public stable release of Az.Support module for the stable version of Support RP API  2020-04-01
-  There are minor differences between this and the preview release. These differences are
    1. Update-AzSupportTicket now supports updating the status of support ticket.
    2. A new severity HighestCriticalImpact was added.
    3. ProductionOutage property from support ticket details object (PSSupportTicket object) was deprecated.
    4. A new property ResourceTypes was added to the response (PSSupportService object) of Get-AzSupportService cmdlet 
- The design document for the preview module is [here ](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/459)
- Resource completer added for Get-AzSupportService and Get-AzSupportProblemClassification cmdlets as per discussion with Mark Cowlinshaw
- Updated help files for all the 4 minor changes mentioned above
- Added new tests

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Support/Support/ChangeLog.md [hunk 1]
```diff
@@ -20,6 +20,17 @@
 
 ## Upcoming Release
 
+## Version 1.0.0
```
Comment (isra-fel) [inline src/Support/Support/ChangeLog.md]: remove this line

File: src/Support/Support/help/Get-AzSupportService.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ Get-AzSupportService -Id <String> [-DefaultProfile <IAzureContextContainer>] [<C
 ```
 
 ## DESCRIPTION
-Gets the current list of Azure services for which support is available. Each Azure service has its own set of categories called problem classification. Get the current list of problem classification for a service using Get-AzSupportProblemClassification. You can use the service and problem classification GUID to create a new support ticket using New-AzSupportTicket.
+Gets the current list of Azure services for which support is available. Each Azure service may contain one of more ARM ResouceTypes that it is responsible for, that you can use to identify the service when creating a technical support ticket for an ARM resource. Each Azure service has its own set of categories called problem classification. Get the current list of problem classification for a service using Get-AzSupportProblemClassification. You can use the service and problem classification GUID to create a new support ticket using New-AzSupportTicket.
```
Comment (ganganarayanan) [inline src/Support/Support/help/Get-AzSupportService.md]: Gets the current list of Azure services for which support is available. Each service may contain one or more Azure resource manager (ARM) resource type information. Resource types (example: 'microsoft.compute/virtualmachines') can be useful to map the right product and ARM resource when creating a technical support ticket.

File: src/Support/Support/help/Update-AzSupportTicket.md [hunk 3]
```diff
@@ -38,13 +38,13 @@ Update-AzSupportTicket -InputObject <PSSupportTicket> [-Severity <Severity>] [-C
...
 ```
-Update-AzSupportTicket -InputObject <PSSupportTicket> [-Severity <Severity>]
+Update-AzSupportTicket -InputObject <PSSupportTicket> [-Severity <Severity>] [-Status <Status>]
  -CustomerContactDetail <PSContactProfile> [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
  [<CommonParameters>]
 ```
 
 ## DESCRIPTION
-Use this cmdlet to update a support ticket's severity level or customer contact details. Note that updating a support ticket's severity level is not allowed when the ticket is assigned to a support engineer. If you wish to update the severity level after ticket assignment, contact the support engineer by sending a communication on the ticket.
+Use this cmdlet to update a support ticket's severity level, status or customer contact details. Note that updating a support ticket's severity level or status is not allowed when the ticket is assigned to a support engineer. If you wish to update the severity level or status after ticket assignment, contact the support engineer by sending a communication on the ticket.
```
Comment (ganganarayanan) [inline src/Support/Support/help/Update-AzSupportTicket.md]: "If you wish to update the severity level or status when an engineer is actively working on your ticket, send a communication on the ticket to the support engineer and request for the severity/status update"

File: src/Support/Support/help/Get-AzSupportService.md (hunk 2 above)
Comment (shahbj79) [reply src/Support/Support/help/Get-AzSupportService.md]: Updated

File: src/Support/Support/ChangeLog.md (hunk 1 above)
Comment (shahbj79) [reply src/Support/Support/ChangeLog.md]: Removed

Comment (shahbj79) [reply src/Support/Support/ChangeLog.md]: Also reverted the change to AssemblyInfo.cs, Az.Support.psd1

**Comment by isra-fel** [comment]
Hi, please read our [GA process doc](https://github.com/Azure/azure-powershell-pr/wiki/Process-of-GA-module) and send us an email to confirm your GA request. Thanks.

**Pull request merged**